pilot/pkg/model/service.go

```diff
@@ -1016,6 +1024,12 @@ type ServiceInfo struct {
 	// Source is the type that introduced this service.
 	Source   TypedObject
 	Waypoint WaypointBindingStatus
+	// MarshaledAddress contains the pre-marshaled representation.
+	// Note: this is an Address -- not a Service.
+	MarshaledAddress *anypb.Any
+	// AsAddress contains a pre-created AddressInfo representation. This ensures we do not need repeated conversions on
+	// the hotpath
+	AsAddress AddressInfo
```

Comment:
does it have duplicate anypb.Any with `MarshaledAddress`

Reply:
One is the marshalled form and one is the unmarshaled form. They are used for different code path